x-pack/plugins/ml/public/jobs/components/custom_url_editor/editor.js

```diff
@@ -0,0 +1,317 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+
+/*
+ * React component for the form for editing a custom URL.
+ */
+
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import {
+  EuiComboBox,
+  EuiFieldText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiForm,
+  EuiFormRow,
+  EuiRadioGroup,
+  EuiSelect,
+  EuiSpacer,
+  EuiTextArea,
+  EuiTitle,
+} from '@elastic/eui';
+
+import {
+  isValidCustomUrlLabel,
+  isValidCustomUrlSettingsTimeRange
+} from 'plugins/ml/jobs/components/custom_url_editor/utils';
+
+import './styles/main.less';
+
+import {
+  TIME_RANGE_TYPE,
+  URL_TYPE
+} from './constants';
+
+function getLinkToOptions() {
+  return [{
+    id: URL_TYPE.KIBANA_DASHBOARD,
+    label: 'Kibana dashboard',
+  }, {
+    id: URL_TYPE.KIBANA_DISCOVER,
+    label: 'Discover',
+  }, {
+    id: URL_TYPE.OTHER,
+    label: 'Other',
+  }];
+}
+
+function onLabelChange(e, customUrl, setEditCustomUrl) {
```

Comment:
I think these functions would be better inside `CustomUrlEditor` and it converted to a class.
Only because you're having to pass `setEditCustomUrl ` in to store the data.
i know we discussed this but i didn't realise the plan was to pass in the function which is called to do store the data.
but i'm happy to go with the majority on this.

Reply:
This is a good candidate for testing out the React Context API, to avoid having to pass down `customUrl` and `setEditCustomUrl` several levels through Props. Will plan to check that out in a later PR.